lib/cfg-grammar.y

```diff
@@ -971,13 +971,15 @@ nonnegative_integer
         : LL_NUMBER
           {
             CHECK_ERROR(($1 >= 0), @1, "It cannot be negative");
+            CHECK_ERROR((($1 >= G_MININT32) && ($1 <= G_MAXINT32)), @1, "Must fit into 32 bits!");
```

Comment:
1)
The documentation does not specify the representation just says `number`, because of that a `32 bits` size of data does not specify a range for the possible values.
I would suggest to write the valid range and the fact that it is integer not float.
`Must be an integer between 0 and G_MAXINT32`.

2)
The `G_MININT32` check is redundant of the previous check, if you want to make sure that `G_MININT32 <= 0` that should be a compile time check instead of runtime.

Reply:
Regarding the message: would `Must be an integer between 0 and 2^31` be acceptable? I do not want to refer to `G_MAXINT32`, nor put its exact value there, because that's hard to quickly parse with the human eye.

Will adjust the minimum check (by removing it).

Reply:
Personally I would still prefer to write the evaluated version, but `2^31` is also acceptable for me.